### ISSUE TITLE
[Vulkan] Invalid use of _MAX_ENUM

### ISSUE THREAD

**Issue by skaarj1989** [wolfpld/tracy#306]
https://github.com/wolfpld/tracy/blob/b83cda575a3f650752e3d35cd8263c8ae8552e1a/TracyVulkan.hpp#L71

https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/chap3.html#fundamentals-validusage-enums

**Comment by wolfpld** [comment]
Yes, and?

**Comment by skaarj1989** [comment]
I'm using [glad](https://github.com/Dav1dde/glad) which uses Vulkan specs ([vk.xml](https://github.com/KhronosGroup/Vulkan-Docs/blob/main/xml/vk.xml)) to generate a loader, and because `_MAX_ENUM` values are not defined there, `TracyVulkan.hpp` gives a compile error.

**Comment by wolfpld** [comment]
Fixed in 69c635f8 (will be pushed later).

```diff
diff --git a/TracyVulkan.hpp b/TracyVulkan.hpp
index e7a19ca1..447f9736 100644
--- a/TracyVulkan.hpp
+++ b/TracyVulkan.hpp
@@ -68,7 +68,7 @@ public:
             if( num > 4 ) num = 4;
             VkTimeDomainEXT data[4];
             _vkGetPhysicalDeviceCalibrateableTimeDomainsEXT( physdev, &num, data );
-            VkTimeDomainEXT supportedDomain = VK_TIME_DOMAIN_MAX_ENUM_EXT;
+            VkTimeDomainEXT supportedDomain = (VkTimeDomainEXT)-1;
 #if defined _WIN32
             supportedDomain = VK_TIME_DOMAIN_QUERY_PERFORMANCE_COUNTER_EXT;
 #elif defined __linux__ && defined CLOCK_MONOTONIC_RAW

```

**Comment by skaarj1989** [comment]
Thanks a lot!
